plugins/ua_nodestore_binary_encoded.c

```diff
@@ -0,0 +1,1163 @@
+    case UA_NODECLASS_OBJECT:
+        retval |= objectNodeEncode((const UA_ObjectNode*)node, bufPos, bufEnd);
+        break;
+    case UA_NODECLASS_VARIABLE:
+        retval |= variableNodeEncode((const UA_VariableNode*)node, bufPos, bufEnd);
+        break;
+    case UA_NODECLASS_METHOD:
+        retval |= methodNodeEncode((const UA_MethodNode*)node, bufPos, bufEnd);
+        break;
+    case UA_NODECLASS_OBJECTTYPE:
+        retval |= objectTypeNodeEncode((const UA_ObjectTypeNode*)node, bufPos, bufEnd);
+        break;
+    case UA_NODECLASS_VARIABLETYPE:
+        retval |= variableTypeNodeEncode((const UA_VariableTypeNode*)node, bufPos, bufEnd);
+        break;
+    case UA_NODECLASS_REFERENCETYPE:
+        retval |= ReferenceTypeNodeEncode((const UA_ReferenceTypeNode*)node, bufPos, bufEnd);
+        break;
+    case UA_NODECLASS_DATATYPE:
+        retval |= dataTypeNodeEncode((const UA_DataTypeNode*)node, bufPos, bufEnd);
+        break;
+    case UA_NODECLASS_VIEW:
+        retval |= viewNodeEncode((const UA_ViewNode*)node, bufPos, bufEnd);
+        break;
+    default:
+        break;
+    }
+
+    return retval;
+}
+#endif
+
+#ifdef UA_ENABLE_USE_ENCODED_NODES
+#define MINIMALNODECOUNT        45 // Total number of nodes in MINIMAL configuration
+const UA_UInt32 minimalNodeIds[MINIMALNODECOUNT] =
+    {UA_NS0ID_REFERENCES, UA_NS0ID_HASSUBTYPE, UA_NS0ID_AGGREGATES, UA_NS0ID_HIERARCHICALREFERENCES,
+     UA_NS0ID_NONHIERARCHICALREFERENCES, UA_NS0ID_HASCHILD, UA_NS0ID_ORGANIZES, UA_NS0ID_HASEVENTSOURCE,
+     UA_NS0ID_HASMODELLINGRULE, UA_NS0ID_HASENCODING, UA_NS0ID_HASDESCRIPTION,UA_NS0ID_HASTYPEDEFINITION,
+     UA_NS0ID_GENERATESEVENT, UA_NS0ID_HASPROPERTY, UA_NS0ID_HASCOMPONENT, UA_NS0ID_HASNOTIFIER,
+     UA_NS0ID_HASORDEREDCOMPONENT, UA_NS0ID_BASEDATATYPE, UA_NS0ID_BASEVARIABLETYPE, UA_NS0ID_BASEDATAVARIABLETYPE,
+     UA_NS0ID_PROPERTYTYPE, UA_NS0ID_BASEOBJECTTYPE, UA_NS0ID_FOLDERTYPE, UA_NS0ID_ROOTFOLDER,
+     UA_NS0ID_OBJECTSFOLDER, UA_NS0ID_TYPESFOLDER, UA_NS0ID_REFERENCETYPESFOLDER, UA_NS0ID_DATATYPESFOLDER,
+     UA_NS0ID_VARIABLETYPESFOLDER, UA_NS0ID_OBJECTTYPESFOLDER, UA_NS0ID_EVENTTYPESFOLDER, UA_NS0ID_VIEWSFOLDER,
+     UA_NS0ID_SERVER, UA_NS0ID_SERVER_SERVERARRAY, UA_NS0ID_SERVER_NAMESPACEARRAY, UA_NS0ID_SERVER_SERVERSTATUS,
+     UA_NS0ID_SERVER_SERVERSTATUS_CURRENTTIME, UA_NS0ID_SERVER_SERVERSTATUS_STATE, UA_NS0ID_SERVER_SERVERSTATUS_BUILDINFO,
+     UA_NS0ID_SERVER_SERVERSTATUS_BUILDINFO_PRODUCTURI, UA_NS0ID_SERVER_SERVERSTATUS_BUILDINFO_MANUFACTURERNAME,
+     UA_NS0ID_SERVER_SERVERSTATUS_BUILDINFO_PRODUCTNAME, UA_NS0ID_SERVER_SERVERSTATUS_BUILDINFO_SOFTWAREVERSION,
+     UA_NS0ID_SERVER_SERVERSTATUS_BUILDINFO_BUILDNUMBER, UA_NS0ID_SERVER_SERVERSTATUS_BUILDINFO_BUILDDATE
+    };
+
+UA_Boolean isMinimalNodesAdded = UA_FALSE;
+#define OFFSET_MEMORYSIZE    2024  // Offset memory size for the edited encode nodes
+#define MAX_ROW_LENGTH         30  // Maximum length of a row in lookup table
+
+lookUpTable *ltRead;
```

Comment:
Global variables should be avoided in the library.
A user might instantiate more than one server in the same process.